app/assets/javascripts/application.js

```diff
@@ -375,7 +375,7 @@ $(function() {
 
 function update_puppetclasses(element) {
   var host_id = $("form").data('id')
-  var env_id = $('*[id*=environment_id]').val();
+  var env_id = $('*[name*=environment_id]').val();
```

Comment:
well, I'm not sure.
```id``` is more accurate when dealing with css and javascript, but ```name``` is the actual field (and values) coming from the server (and shipped to the server) so it is accurate as much as ```id```. 
In Rails, they would usually be with the same name, afaik. So I think it is the correct selector.
Also, ```id``` may be changed by other components (javascript, deface) for some reason - name can not be changed as it will compromise submitting the data back to the server